server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -723,6 +731,30 @@ Map<Index, List<IndexShardStats>> statsByShard(final IndicesService indicesServi
         return statsByShard;
     }
 
+    Map<Index, CommonStats> statsByIndex(final IndicesService indicesService, final CommonStatsFlags flags) {
+        final Map<Index, CommonStats> statsByIndex = new HashMap<>();
+
+        for (final IndexService indexService : indicesService) {
+            for (final IndexShard indexShard : indexService) {
+                try {
+                    final IndexShardStats indexShardStats = indicesService.indexShardStats(indicesService, indexShard, flags);
+                    if (indexShardStats == null) {
+                        continue;
+                    }
+                    if (!statsByIndex.containsKey(indexShard.shardId().getIndex())) {
+                        statsByIndex.put(indexShard.shardId().getIndex(), new CommonStats());
+                    }
+                    Arrays.stream(indexShardStats.getShards())
+                        .forEach(shardStats -> statsByIndex.get(indexShard.shardId().getIndex()).add(shardStats.getStats()));
+                } catch (IllegalIndexShardStateException | AlreadyClosedException e) {
+                    // we can safely ignore illegal state on ones that are closing for example
+                    logger.trace(() -> new ParameterizedMessage("{} ignoring shard stats", indexShard.shardId()), e);
+                }
```

Comment:
In the try block we are performing 2 items

1. Perform fetch via indexShardStats operation and handle corresponding exceptions within it.
2. Perform update in the map statsByIndex.

Can we decouple these 2 code blocks or even delegate work to 2 independent private methods e..g  getIndexShardStats & updateStatsForIndex so that its clear and mapped to logical blocks.